presto-tests/src/test/java/com/facebook/presto/tests/TestRestSqlFunctionExecutor.java

```diff
@@ -0,0 +1,262 @@
+
+        restSqlFunctionExecutor = createRestSqlFunctionExecutor(testingHttpClient);
+    }
+
+    private RestSqlFunctionExecutor createRestSqlFunctionExecutor(HttpClient testingHttpClient)
+    {
+        Bootstrap app = new Bootstrap(
+                // Specially use a testing HTTP client instead of a real one
+                binder -> binder.bind(HttpClient.class).annotatedWith(RestClient.class).toInstance(testingHttpClient),
+                binder -> binder.bind(NodeManager.class).toInstance(createNodeManager()),
+                // Otherwise use the exact same module as the rest sql function executor service
+                new SimpleAddressRestSqlFunctionExecutorsModule());
+
+        Injector injector = app
+                .doNotInitializeLogging()
+                .setRequiredConfigurationProperties(
+                        ImmutableMap.of(
+                                "rest-uri", REST_SERVER_URI.toString())
+                ).initialize();
+        return injector.getInstance(RestSqlFunctionExecutor.class);
+    }
+
+    //@Path("/v1/functions/{schema}/{functionName}/{functionId}/{version}")
+    //@Path("v1/functions/abs?returnType=integer&numArgs=1&argType1=integer")
+    @Path("/v1/functions/{functionName}")
+    public static class TestingFunctionResource
+    {
+        private final PagesSerde pagesSerde;
+        public TestingFunctionResource(PagesSerde pagesSerde)
+        {
+            this.pagesSerde = requireNonNull(pagesSerde);
+        }
+
+        @POST
+        @Consumes(MediaType.WILDCARD)
+        @Produces(MediaType.WILDCARD)
+        public byte[] post(
+                @PathParam("functionName") String functionName,
+                @QueryParam("returnType") String returnType,
+                @QueryParam("numArgs") int numArgs,
+                byte[] serializedPageByteArray)
+        {
+            Slice slice = wrappedBuffer(serializedPageByteArray);
+            SerializedPage serializedPage = readSerializedPage(new BasicSliceInput((slice)));
+            Page inputPage = pagesSerde.deserialize(serializedPage);
+            List<Type> types = new ArrayList<>();
+            types.add(IntegerType.INTEGER);
+            List<Object> values = new ArrayList<>();
+            if (functionName.equals("power_tower")) {
+                values.add((long) Math.pow(inputPage.getBlock(0).toLong(0), inputPage.getBlock(0).toLong(0)));
+            }
+            if (functionName.equals("abs")) {
+                values.add(Math.abs(inputPage.getBlock(0).toLong(0)));
+            }
+            Page outputPage = createPage(types, values);
+            DynamicSliceOutput sliceOutput = new DynamicSliceOutput((int) outputPage.getRetainedSizeInBytes());
+            writeSerializedPage(sliceOutput, pagesSerde.serialize(outputPage));
+            return sliceOutput.slice().byteArray();
+        }
+    }
+
+    @Test(dataProvider = "testExecuteFunctionAbs")
```

Comment:
Let's add test cases for all of the types that are supported in InterpretedFunctionInvoker.  E.g. INTEGER, DOUBLE, Slice, etc.